### PR TITLE
fixes #14513 - add a rss feed dashboard widget

### PR DESCRIPTION
**Pull request by stbenjam**
Displays the most recent 8 items of an RSS feed.  I actually had this sitting around for a while and just cleaned it up a bit.  I think it'd be cool to have a widget that shows the Foreman blog.

~~Unfortunately the Jekyll blog RSS isn't liked at all by Ruby's RSS built-in thingy (http://theforeman.org/feed.xml).  When I worked on this before, the Blogspot RSS was working with it.~~

The ruby built-in RSS reader doesn't seem to support Atom, so I'm pulling in the simple-rss gem.  There's a bunch of others that seem adequate, but this is what ManageIQ is using, and it seems to work just fine.

widget

Thoughts?

### CODE REVIEW

**Comment by GregSutcliffe** [comment]
I'll stay out of the review, but +1 for the idea, I like it!

**Comment by dLobatog** [comment]
Very cool in my opinion too! It looks easy enough to extract to a plugin, but to me it looks like an interesting idea we should offer by default. A good compromise here might be to make it a plugin, but ship it enabled by default on the installer. Sounds good? cc @ohadlevy @domcleal @ares

File: app/services/latest_news.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @url = url
+    @date = date
+  end
+
+  class << self
+    attr_reader :last_updated
+
+    def items
+      if @last_updated && (Time.now.utc < (@last_updated + 30.minutes))
+        return @items
```
Comment (dLobatog) [inline app/services/latest_news.rb:17]: No need for these 2 returns when they're the last statement

File: test/unit/latest_news_test.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+require 'test_helper'
+
+class LatestNewsTest < ActiveSupport::TestCase
+  setup do
+    Setting[:latest_news_rss_feed] = File.expand_path(File.dirname(__FILE__) + '/rss_feed.xml')
```
Comment (dLobatog) [inline test/unit/latest_news_test.rb:5]: I don't think testing SimpleRSS parses the file & fetches items is our responsibility here, they already test that. https://github.com/cardmagic/simple-rss/blob/master/test/base/base_test.rb

Why not just expecting the `.parse` calls, which is really all that Foreman is concerned about?

**Comment by ares** [comment]
My 2c - I'd prefer to have this in core actually. I don't see much benefit of having it in plugin which is enabled by default. It does not add much code to maintain. Maintaining another plugin (testing, packaging, reviews) seems like too much effort to me.

**Comment by timogoebel** [comment]
When phoning home, the connection might be blocked by a corporate firewall (which does not necessarily reject the connection, but drops it resulting in a timeout) in enterprise environments. Would that delay page rendering? Should we add a global setting to disable any phone home calls?

File: test/unit/latest_news_test.rb (hunk 2 above)
Comment (stbenjam) [inline test/unit/latest_news_test.rb:5]: Well, sure, it's indirectly testing the behavior but stubs will be more complicated.  I test the LatestNews service returns the right number of items, caches them for the specified period of time, handles exceptions, etc.  Having an RSS fixture seemed simpler...

**Comment by stbenjam** [comment]
@timogoebel Indeed, we turned Gravatar off for that very reason.  

I'm happy to extract this to a plugin, and users can enable it if they want it.  It could be advertised from the blog, perhaps.  If we don't want a widget reaching out to the internet by default, then it might as well be a plugin. No sense in adding it to core if it's just off by default.

**Comment by tbrisker** [comment]
+1 to doing this in a plugin. This is cool functionality to have but I'm not sure it has a place in core, esp. considering some (many?) instances are blocked from the internet. Also - it would be nice if the plugin would you add more feeds to the dashboard according to your preferences.

**Comment by lzap** [comment]
Haha nice feature:

image

I am for plugin in general, but if we are planning a feture that will leverage RSS library (perhaps providing some RSS channels) then let's push it in the core.

**Comment by stbenjam** [comment]
If someone wants to create a repo for me in the foreman org, I'll move this to a plugin.

Let's call it, `foreman_news`?

**Comment by dLobatog** [comment]
If it's in core (I think it'd be so helpful for users), how about having the dashboard widget added by default? 

Since many people seem to be concerned about contacting the outside by default, my proposal is that the widget should have a text "Click enable to see the latest Foreman news. Notice this will make Foreman contact theforeman.org" and a button to enable/disable it. The button is clickable only for Foreman admins (or anyone who can change settings). Sounds good? @GregSutcliffe @ares @timogoebel @lzap @tbrisker

**Comment by stbenjam** [comment]
That's a pretty good solution, easier than maintaining the overhead of a plugin.  If it sounds ok, I can re-open and make the necessary changes

**Comment by GregSutcliffe** [comment]
@dLobatog unsurprisingly, I am +1 for that :)

**Comment by stbenjam** [comment]
Alright, I'll reopen it and try out @dLobatog's suggestion when I get a minute

**Pull request closed without merging**